### PR TITLE
Use gcc crate in native build script example

### PR DESCRIPTION
**Pull request by jdub**

### CODE REVIEW

File: src/doc/build-script.md [hunk 1]
```diff
@@ -350,16 +350,16 @@ portable, and standardized. For example, the build script could be written as:
 ```rust,ignore
 // build.rs
 
-// Bring in a dependency on an externally maintained `cc` package which manages
+// Bring in a dependency on an externally maintained `gcc` package which manages
 // invoking the C compiler.
 extern crate cc;
```
Comment (alexcrichton) [inline src/doc/build-script.md]: I think this'll need to be `gcc`

File: src/doc/build-script.md [hunk 2]
```diff
@@ -350,16 +350,16 @@ portable, and standardized. For example, the build script could be written as:
...
 
 fn main() {
-    cc::compile_library("libhello.a", &["src/hello.c"]).unwrap();
+    gcc::compile_library("libhello.a", &["src/hello.c"]).unwrap();
 }
 ```
 
-This example is a little hand-wavy, but we can assume that the `cc` crate
+This example is a little hand-wavy, but we can assume that the `gcc` crate
 performs tasks such as:
```
Comment (alexcrichton) [inline src/doc/build-script.md]: Could you rephrase this to link to the gcc crate on crates.io?

**Comment by jdub** [comment]
Fixed the executable bit + squashed.

File: src/doc/build-script.md [hunk 3]
```diff
@@ -350,17 +350,25 @@ portable, and standardized. For example, the build script could be written as:
...
 ```
 
-This example is a little hand-wavy, but we can assume that the `cc` crate
-performs tasks such as:
+Add a build time dependency on the `gcc` crate with the following addition to
+your `Cargo.toml`:
+
+```toml
+[build-dependencies]
+gcc = "*"
```
Comment (alexcrichton) [inline src/doc/build-script.md]: Could you switch to 0.3 instead of *? Other than that looks good to me!

Comment (jdub) [inline src/doc/build-script.md]: Done!
